sw/device/silicon_creator/mask_rom/mask_rom_epmp.c

```diff
@@ -0,0 +1,74 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/mask_rom/mask_rom_epmp.h"
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/csr.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// Symbols defined in linker script.
+extern char _stack_start[];  // Lowest stack address.
+extern char _text_start[];   // Start of executable code.
+extern char _text_end[];     // End of executable code.
+
+void mask_rom_epmp_state_init(epmp_state_t *state) {
```

Comment:
Will this state be shared between boot stages or will each stage have a `*_epmp_state_init()` function?

Reply:
IMO `ROM_EXT` at least should setup the ePMP state itself. It should have enough information to regenerate it without needing to pass the state around.

I'm not sure about the other boot stages.

Reply:
Each stage will need its own I think. There may be some overlap but in theory the other boot stages could use a totally different schema if they wanted to.

Reply:
SG. Since boot stages are decoupled thanks to ePMP, I think it's better/cleaner to have separate state init functions.